### PR TITLE
Feature/issue 1012 binorm copula cdf

### PR DESCRIPTION
**Pull request by rtrangucci**
## Summary

Adds `binormal_copula_cdf` to the math library.

## Tests

I've written tests for prim, rev, fwd, and mix, which test gradients and higher-order derivatives.
* `test/unit/math/prim/mat/prob/binormal_copula_cdf_test.cpp`
* `test/unit/math/rev/mat/prob/binormal_copula_cdf_test.cpp`
* `test/unit/math/fwd/mat/prob/binormal_copula_cdf_test.cpp`
* `test/unit/math/mix/mat/prob/binormal_copula_cdf_test.cpp`

## Side Effects

No side effects

## Checklist

- [x] Math issue #1012 

- [x] Copyright holder: Columbia University

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [ ] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

**Comment by rtrangucci** [comment]
@bgoodri this is ready for review when you get a chance!

File: stan/math/prim/mat/prob/binormal_copula_cdf.hpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ * @return The probability P(U1 <= u[1], U2 <= u[2] | rho).
+ * @throw std::domain_error if the rho is not between -1 and 1 or nan.
+ * @throw std::domain_error if elements of u aren't between 0 and 1 or nan.
+ * @throw std::invalid_argument u isn't size 2.
+ * @tparam T_u Type of random variable.
+ * @tparam T_rho Type of correlation.
+ */
+
+template <typename T_u, typename T_rho>
+typename return_type<T_u, T_rho>::type binormal_copula_cdf(const T_u& u,
```
Comment (bgoodri) [inline stan/math/prim/mat/prob/binormal_copula_cdf.hpp]: Should we be calling this `std_binormal_copula_cdf` then?

Comment (bgoodri) [reply stan/math/prim/mat/prob/binormal_copula_cdf.hpp]: Or I guess people consider the location and scale to be part of the marginal distribution

Comment (rtrangucci) [reply stan/math/prim/mat/prob/binormal_copula_cdf.hpp]: Yeah, that was what I was thinking. I haven't seen the bivariate normal copula as having a nonstandard form.

**Comment by syclik** [comment]
@bgoodri, mind finishing out this review? If not, kick it back to me and I'll find someone to review it.

**Comment by syclik** [comment]
Closing due to inactivity.

**Pull request closed without merging**